benchmark/conftest.py

```
@@ -165,12 +167,21 @@ def pytest_configure(config):
             for arg in config.invocation_params.args
         ]
 
-        logging.basicConfig(
-            filename="result_{}.log".format("_".join(cmd_args)).replace("_-", "-"),
-            filemode="w",
-            level=logging.INFO,
-            format="[%(levelname)s] %(message)s",
-        )
+        log_file = "result_{}.log".format("_".join(cmd_args)).replace("_-", "-")
+
+        for h in list(recordLogger.handlers):
+            recordLogger.removeHandler(h)
+            try:
+                h.close()
+            except Exception:
+                pass
```

pass?
should we expose the reason why exception?

---

This logic is for cleanup (removing stale handlers before adding a new one).

If h.close() fails (e.g., file already closed or permission issue on an old handle), we generally want to ignore it because:

1. It shouldn't block the new test run from starting.
2. We are removing this handler from the logger anyway ( removeHandler is called right before).
However, you are right that a silent pass might hide underlying issues. I have updated the code to raise a warning ( warnings.warn ) instead of silently passing, so we can be aware if something goes wrong while still keeping the cleanup process robust.